crates/hyperion/src/simulation/inventory.rs

```diff
@@ -1191,6 +1208,74 @@ fn handle_drop_key(
     event_writer.write(event);
 }
 
+fn validate_slot_changes(
+    slot_changes: &[SlotChange],
+    inventories_mut: &[&mut ItemSlot],
+    cursor_item: &ItemStack,
+    player_only: bool,
+) -> bool {
+    // If cursor is empty, no slot changes should be valid
+    if cursor_item.is_empty() {
+        return false;
+    }
+
+    for slot_change in slot_changes {
+        // Validate slot index bounds
+        let Ok(slot_idx) = usize::try_from(slot_change.idx) else {
+            return false;
+        };
+
+        // Check if slot exists in our inventory
+        if slot_idx >= inventories_mut.len() {
+            return false;
+        }
+
+        let slot = &inventories_mut[slot_idx];
```

Comment:
The manual bounds check can be removed here:
```suggestion
        let Some(slot) = inventories_mut.get(slot_idx) else {
            return false;
        }
```